kong/runloop/plugins_iterator.lua

```diff
@@ -97,17 +97,24 @@ local function get_next(self)
 
   -- load the plugin configuration in early phases
   if self.access_or_cert_ctx then
+    local schema = plugin.schema or {}
 
     local api          = self.api
     local route        = self.route
     local service      = self.service
     local consumer     = ctx.authenticated_consumer
 
-    if consumer then
-      local schema = plugin.schema
-      if schema and schema.no_consumer then
-        consumer = nil
-      end
+    if api and schema.no_api then
+      api = nil
+    end
+    if route and schema.no_route then
+      route = nil
+    end
+    if service and schema.no_service then
+      service = nil
+    end
+    if consumer and schema.no_consumer then
+      consumer = nil
```

Comment:
I see that the original code had the same checks and balances, but isn't this a migration problem? Those cases should not exist, so if they do, something else failed.

If the code stays, it should generate warnings maybe?

@thibaultcha what do you think?

Reply:
> Those cases should not exist

Which cases are you referring to?

Reply:
My thought was: if a plugin schema has `no_route == true` then we should not need to set `route = nil` since it already should be `nil`.

rereading the code it seems that `route` holds the current request-route, not the configured one for the plugin.

Sorry for the noise.